lib/cooked_post_processor.rb

```diff
@@ -382,8 +382,12 @@ def add_lightbox!(img, original_width, original_height, upload, cropped: false)
     a = create_link_node("lightbox", img["src"])
     img.add_next_sibling(a)
 
-    if upload && Discourse.store.internal?
-      a["data-download-href"] = Discourse.store.download_url(upload)
+    if upload
+      if Discourse.store.internal?
+        a["data-download-href"] = Discourse.store.download_url(upload)
```

Comment:
wait ... why can't we trust "download_url" to always be correct regardless of store? "dl=1" to me feels very much like a "store" concern not something cooked_post_processor.

Reply:
You're right of course, I've made the change, it's cleaner to have this in the store.